src/vs/workbench/services/workingCopy/common/workingCopyFileService.ts

```diff
@@ -57,15 +64,34 @@ export interface IWorkingCopyFileOperationParticipant {
 	 * change the working copies before they are being saved to disk.
 	 */
 	participate(
-		files: SourceTargetPair[],
+		files: SourceTargetPairWithOptions[],
 		operation: FileOperation,
-		undoRedoGroupId: number | undefined,
-		isUndoing: boolean | undefined,
 		timeout: number,
 		token: CancellationToken
 	): Promise<void>;
 }
 
+export interface IDeleteOperation {
+	resource: URI;
+	options?: {
+		useTrash?: boolean;
+		recursive?: boolean;
+		undoRedoGroupId?: number;
```

Comment:
I wonder if there is a way to extract a generic interface for `undoRedoGroupId` and `isUndoing` and then use it for all operations and the `SourceTargetPairWithOptions`?